crates/configuration/src/configuration.rs

```diff
@@ -4,13 +4,32 @@ use anyhow::{anyhow, ensure};
 use itertools::Itertools;
 use mongodb_support::BsonScalarType;
 use ndc_models as ndc;
+use serde::Deserialize;
 
 use crate::{
     native_procedure::NativeProcedure,
     native_query::{NativeQuery, NativeQueryRepresentation},
     read_directory, schema, serialized,
 };
 
+#[derive(Clone, Debug, Deserialize)]
+pub struct ConfigurationOptions {
```

Comment:
Do we want some more structure in this so we can add extra sections in the future? The current options could go under an `introspection_options` key, that would leave from for other options in the future, such as for DB connection settings. E.g.
```yaml
introspection_options:
    sample_size: 100
    no_validator_schema: false
connection_options:
    whatever
...
```